src/Plugin/Field/FieldType/ExecutablePluginItem.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\commerce\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\TypedData\MapDataDefinition;
+
+/**
+ * Plugin implementation of the 'commerce_executable_plugin_item' field type.
+ *
+ * @FieldType(
+ *   id = "commerce_executable_plugin_item",
+ *   label = @Translation("Executable plugin"),
+ *   description = @Translation("Stores configuration for an executable plugin."),
+ *   default_widget = "commerce_executable_plugin_select",
+ *   default_formatter = "string",
+ *   deriver = "\Drupal\commerce\Plugin\Field\FieldType\ExecutablePluginItemDeriver"
+ * )
+ */
+class ExecutablePluginItem extends FieldItemBase implements ExecutablePluginItemInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function propertyDefinitions(FieldStorageDefinitionInterface $field_definition) {
+    $properties['executable_id'] = DataDefinition::create('string')
+      ->setLabel(t('Executable ID'));
+    $properties['executable_configuration'] = MapDataDefinition::create()
+      ->setLabel(t('Executable configuration'));
+
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function schema(FieldStorageDefinitionInterface $field_definition) {
+    return [
+      'columns' => [
+        'executable_id' => [
+          'description' => 'The executable ID.',
+          'type' => 'varchar_ascii',
+          'length' => 255,
+          'not null' => TRUE,
+        ],
+        'executable_configuration' => [
+          'description' => 'The executable configuration.',
+          'type' => 'blob',
+          'not null' => TRUE,
+          'serialize' => TRUE,
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function mainPropertyName() {
+    return 'executable_id';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isEmpty() {
+    return empty($this->executable_id) || $this->executable_id == '_none';
+  }
+
+  /**
+   * Gets the exectuable plugin's manager.
+   *
+   * @return \Drupal\Core\Executable\ExecutableManagerInterface
+   *   The plugin manager.
+   */
+  public function getExecutableManager() {
+    return \Drupal::service('plugin.manager.' . $this->getPluginDefinition()['plugin_type_id']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInstance(array $contexts = []) {
+    $plugin = $this->getExecutableManager()
+      ->createInstance($this->executable_id, $this->executable_configuration);
+
+    // Just because the plugin is context aware, we cannot gurantee the
+    // plugin manager sets them. So we apply the context mapping as well.
+    \Drupal::service('context.handler')
+      ->applyContextMapping($plugin, $contexts);
```

Comment:
It's weird that we broke this line into two. Either have it be one line, or assign the context handler to a variable first.